db/migrate/1599665760_add_time_and_weeks_to_timetable_slots.rb

```diff
@@ -0,0 +1,9 @@
+Sequel.migration do
+  change do
+    alter_table(:timetable_slots) do
+      add_column :start_time, Time, only_time: true, default: nil, null: true
+      add_column :end_time, Time, only_time: true, default: nil, null: true
+      add_column :weeks, 'int4range[]', default: nil, null: true
```

Comment:
I’ve eventually changed my mind about the data type for the `weeks` column. I’ve realized that its domain is just 1–13 (maybe 14), so it will be much easier to work with an expanded array of integers instead of array of ranges. My fault that I didn’t realize it before. :/

Reply:
I'm not sure which solution is a better idea. If you rework it as an expanded array of integers, you still need to provide some date intervals when planning the timetable. 

- You can use intervals by individual weeks separately. OR
- You can merge weeks together.
Both have their own pros and cons.

Or think of a completely different solution on how to intersects week planning over the semester period planning.